code_samples/back_office/menu/menu_item/src/EventSubscriber/MyMenuSubscriber.php

```diff
@@ -5,22 +5,39 @@
 use Ibexa\AdminUi\Menu\Event\ConfigureMenuEvent;
 use Ibexa\AdminUi\Menu\MainMenuBuilder;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Ibexa\AdminUi\Menu\MenuItemFactory;
 
 class MyMenuSubscriber implements EventSubscriberInterface
 {
+    private $menuItemFactory;
+
+    public function __construct(MenuItemFactory $menuItemFactory)
+    {
+        $this->menuItemFactory = $menuItemFactory;
```

Comment:
`ConfigureMenuEvent` contains `Factory` (see `ConfigureMenuEvent::getFactory()`) and so it should not be injected by service constructor. Although my previous comment makes it entirely uneccessary to access the factory directly btw.